### PR TITLE
feat: Calendar View

### PR DESCRIPTION
**Pull request by RitvikSardana**
This PR adds a new Component into the Frappe UI called "Calendar Component".
The story book, the docs and custom events of the component are also added in the PR.

Demo:

https://github.com/frappe/frappe-ui/assets/65544983/8b89d7ba-c61d-4bb9-a5b7-f6270e05e4dc

List of features.

1. Monthly, Weekly, Daily View
2. CRUD operations for Calendar Events.
3. Drag and Drop Functionality in all 3 views.
4. Resize any calendar event
5. Clashing Events handling 
6. Full Day events support.
7. Move event within days or in the same day.
8.  Flexible Component API.
    - Custom API Integration (for CRUD operations on Event).
    - Custom Header.
    - Custom Click events (onClick, onDblClick, onCellDblClick)

### CODE REVIEW

File: src/components/CalendarView/CalendarMonthly.vue [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      >
+    </div>
+
+    <!-- Date Grid -->
+    <div
+      class="grid h-full w-full grid-cols-7 grid-rows-6 border-l-[1px] border-t-[1px]"
+    >
+      <div
+        v-for="date in currentMonthDates"
+        class="h-28 overflow-scroll border-b-[1px] border-r-[1px] border-gray-200"
```
Comment (surajshetty3416) [inline src/components/CalendarView/CalendarMonthly.vue:18]: You might have to remove `overflow-scroll`... causes following issue
Screenshot 2024-05-29 at 4 10 40 PM

File: src/components/CalendarView.story.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    	:events=events
+
+    	:config=config
+
+    	@createEvent=(event) => createEvent(event)
+
+    	@updateEvent=(event) => updateEvent(event)
+
+    	@deleteEvent=(eventID) => deleteEvent(eventID)
```
Comment (surajshetty3416) [inline src/components/CalendarView.story.md:124]: No need to have "Event" in event name

you can directly call `@create` like `@click` or else `onCreate`/`onUpdate` which are more intuitive.

**Review by surajshetty3416 — CHANGES_REQUESTED** [review]

File: src/components/CalendarView/CalendarMonthly.vue (hunk 1 above)
Comment (RitvikSardana) [reply src/components/CalendarView/CalendarMonthly.vue:18]: What is the issue here, did not understand?

Comment (surajshetty3416) [reply src/components/CalendarView/CalendarMonthly.vue:18]: > What is the issue here, did not understand?

Scrollbars in each cell... shouldn't it look like this.
Screenshot 2024-05-30 at 9 49 37 AM

**Pull request merged**